upup/models/cloudup/_aws/network.yaml

```diff
@@ -33,16 +33,19 @@ route/0.0.0.0/0:
   internetGateway: internetGateway/{{ ClusterName }}
   vpc: vpc/{{ ClusterName }}
 
-{{ range $zone := .Zones }}
+{{ range $index, $zone := .Zones }}
 
 subnet/{{ $zone.Name }}.{{ ClusterName }}:
   vpc: vpc/{{ ClusterName }}
   availabilityZone: {{ $zone.Name }}
   cidr: {{ $zone.CIDR }}
   id: {{ $zone.ProviderID }}
+  shared: {{ SharedZone $index }}
 
+{{ if not (SharedZone $index) }}
```

Comment:
I would have guessed you could just pass $zone here but I'll just merge, try it for myself and see what goes wrong :-)